skbio/stats/composition.py

```diff
@@ -1105,6 +1235,65 @@ def _gram_schmidt_basis(n):
     return basis.T
 
 
+def sbp_basis(sbp):
+    """
+    Builds an orthogonal basis from a sequential binary partition (SBP).
+    As explained in [1], the "SBP describes the D−1 orthogonal (geometrically
+    independent) balances between parts and groups of parts. The SBP is a
+    (D - 1) × D matrix, in which parts labeled '+1' (group  numerator) are
+    balanced with parts labeled '−1' (group denominator). A part labeled '0'
+    is excluded from the balance between parts. The composition is partitioned
+    sequentially into contrasts at every hierarchically ordered row until
+    the '+1' and '-1' groups each contain a single part."
```

Comment:
I think it would be great to include a mathematical formula to describe the sequential binary partition.  It may not be intuitive to understand what a `group` is in this context.  But showing how this links to calculating the log ratios between groups could clarify this.

I'm thinking about the classical definition of a balance where

``` 
b_i = sqrt(r*s / r+s) * log(g(x_r) / g(x_s))
```

And explicitly stating that the sequential binary partition denotes numerator and denominator membership for the `ith` log ratio.

What do you think?